src/GraphQL/Execution/ExecutionResultJsonConverter.cs

```diff
@@ -86,7 +85,7 @@ private void WriteErrors(ExecutionErrors errors, JsonWriter writer, JsonSerializ
 
         private void WriteErrorExtensions(ExecutionError error, JsonWriter writer, JsonSerializer serializer)
         {
-            if (string.IsNullOrWhiteSpace(error.Code) && error.Data?.Count == 0)
+            if (string.IsNullOrWhiteSpace(error.Code) && (error.Data == null || error.Data.Count == 0))
```

Comment:
Condition `error.Data?.Count == 0` gives `false` if Data is null because null <> 0, so NRE may occur later